final_task/rss_reader/ClassNews.py

```diff
@@ -0,0 +1,65 @@
+import re
+import html2text
+
+links_template = '\"((http|https)://(\w|.)+?)\"'
+
+
+def xml_arguments_for_class(xml_string, limit):
+    """This function receive the xml and limit of articles and returns list of dictionaries"""
+    dict_article_list = []
+    for counter, neighbor in enumerate(xml_string.iter('item')):
+        parser_dictionary = {}
+        text = html2text.HTML2Text()
+        text.ignore_images = True
+        text.ignore_links = True
+        text.ignore_emphasis = True
+        for child in neighbor:
+            # Here we create the article in the form of a dictionary
+            if child.tag == 'title':
+                parser_dictionary['title'] = text.handle(child.text).replace('\n', "")
+
+            if child.tag == 'pubDate':
+                parser_dictionary['date'] = child.text
+
+            if child.tag == 'link':
+                parser_dictionary['link'] = child.text
+
+            if child.tag == 'description':
+                parser_dictionary['article'] = text.handle(child.text).replace('\n', '')
+
+                # здесь мы ищем все ссылки, их шаблон задан в links_template и находиться ответ будет в группе 1
+                list_links = []
+                for group1 in re.finditer(links_template, child.text):
+                    list_links.append(group1.group(1))
+                    parser_dictionary['links'] = list_links
+                    # Еще добавить описание к картинкам
+        dict_article_list.append(parser_dictionary)
+        if limit == counter + 1:
```

Comment:
If the limit is larger than feed size this function will return `None`

Reply:
As far as I understand, with slicing the list, I would need to parse and build the whole feed list first, and it might be inefficient in case if limit is much less than feed size. But I can change it, sure